src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1434,7 +1448,9 @@ void analyzeSymbol(JCTree tree) {
                     }
                     // Field may not have an initializer
                     if ((sym.flags() & HASINIT) != 0) {
-                        reportPrologueError(tree, sym, true);
+                        if (!sym.type.hasTag(ARRAY) || !isIndexed) {
```

Comment:
Does this work for complex ones like `(this.a = stuff)[b] = 5`?

Maybe we can try updating `isInLHS` to point to a JCTree and all the tests to be something like `tree == isInLHS` so this just won't run for array indexed trees?